### DISCUSSION TITLE
Compiler 'arm-none-eabi-gcc' not found on Pico W build

### DISCUSSION THREAD

**Discussion by ZacharyInDallas** [micropython/micropython#11461] [RP2040 / Pico] [answered]
I’m building MicroPython for Pico W using WSL2:
Ubuntu 20.04.4 LTS (GNU/Linux 5.15.90.1-microsoft-standard-WSL2 x86_64).

After following the instructions on https://github.com/micropython/micropython/tree/master/ports/rp2

I get Compiler 'arm-none-eabi-gcc' not found while building.

**Comment by sosi-deadeye** [comment]
You need a compiler, which can produce binaries for the target.

Maybe it works with the package `gcc-arm-none-eabi`.

```
sudo apt-get install gcc-arm-none-eabi
```

EDIT:
I can only find the Package `gcc-arm-linux-gnueabihf`. I think this does not work with Ubuntu for WSL.

**Reply by ZacharyInDallas** [reply]
Is there any documentation about this?  The instructions for building the port don't mention this as even being needed, much less where to get them.

**Reply by jimmo** [reply]
Yes this is a bit of an oversight in the readme. We do have equivalent instructions for the other ports. I is very difficult to provide "one size fits all" docs for obtaining toolchains that applies to all distros (as well as other operating systems), but yes it should at least say that you need the cortex-m compiler.

In the meantime, the Pico port uses the Pico SDK. Their documentation is here: https://www.raspberrypi.com/documentation/pico-sdk/

**Reply by ZacharyInDallas** [reply, marked as answer]
I found a fuller explanation on how to do the build at:

Raspberry Pi Pico Python SDK
https://datasheets.raspberrypi.com/pico/raspberry-pi-pico-python-sdk.pdf

It mentions:

$ sudo apt update
$ sudo apt install cmake gcc-arm-none-eabi libnewlib-arm-none-eabi build-essential

That worked for me.

**Reply by ZacharyInDallas** [reply]
I would say that this fix should be mentioned in the MicroPython documentation.  That documentation seems to represent itself as being complete.  If you follow it exactly, you can't do a build.

**Comment by cve2022** [comment]
Take a look at #10920 

I believe it must work for 20.04 as well.

HTH
